### PR TITLE
Rwa 1021 - Add functional tests for consuming messages from DLQ & inserting in DB

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1021

### Change description ###

 Add functional tests for consuming messages from DLQ & inserting in DB

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+@Slf4j
+@ActiveProfiles(profiles = {"local", "functional"})
+public class DlqMessagesToDatabaseTest extends MessagingTests {
+
+    @Test
+    public void should_store_dlq_messages_in_database()  {
+        List<String> messageIds = List.of(randomMessageId(), randomMessageId(), randomMessageId());
+
+        var caseId = "1111222233334444";
```
Comment (rafalkalita) [inline src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java:32]: I would generate the caseId, so we don't run into multiple services hitting AAT failing tests issue.
`var caseId = RandomStringUtils.randomNumeric(16);`

File: src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    } else {
+                        return false;
+                    }
+                });
+    }
+
+    @Test
+    public void should_store_dlq_messages_missing_mandatory_fields_in_database_as_unprocessable()  {
+
+        var caseId = "5555666677778888";
```
Comment (rafalkalita) [inline src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java:70]: Same as above. I think random caseId would be better.

Comment (sullivangareth) [reply src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java]: Fixed in [a256fa9](https://github.com/hmcts/wa-case-event-handler/pull/224/commits/a256fa9440cf9d1927c7a4d3ee194fcb5f351adb)

File: src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java (hunk 1 above)
Comment (sullivangareth) [reply src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java]: Fixed in [a256fa9](https://github.com/hmcts/wa-case-event-handler/pull/224/commits/a256fa9440cf9d1927c7a4d3ee194fcb5f351adb)

**Review by rafalkalita — APPROVED** [review]

**Pull request merged**